### PR TITLE
create select package, move stuff from labs over

### PR DESCRIPTION
**Pull request by hellochar**
Graduates the following components into their own package, `@blueprintjs/select`:

* Select
* Suggest
* MultiSelect
* Omnibar
* QueryList

This PR currently fails due to 

- [x] Reference Popover2 and TagInput once they're in core (@cmslewis for SA), waiting on https://github.com/palantir/blueprint/pull/1881
- [x] ensure the docs page works with the new package (and update https://github.com/palantir/blueprint/wiki/Creating-a-new-package accordingly)
- [x] Look through unit tests and see how good we are (run tests with coverage and see if it fails)

### CODE REVIEW

File: packages/select/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<img height="204" src="https://cloud.githubusercontent.com/assets/464822/20228152/d3f36dc2-a804-11e6-80ff-51ada2d13ea7.png">
+
+# [Blueprint](http://blueprintjs.com/) Laboratory for Component Development
```
Comment (adidahiya) [inline packages/select/README.md]: This title needs updating

File: packages/select/package.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  "main": "dist/index.js",
+  "module": "dist/esm/index.js",
+  "typings": "dist/esm/index.d.ts",
+  "style": "dist/blueprint-select.css",
+  "unpkg": "dist/select.bundle.js",
+  "scripts": {
+    "compile": "npm-run-all -p 'compile:*'",
+    "compile:esm": "tsc -p ./src",
+    "compile:cjs": "tsc -p ./src/tsconfig.cjs.json",
+    "compile:css": "node-sass --importer ../../node_modules/node-sass-package-importer/dist/cli.js --recursive src --output dist/ --source-map true",
```
Comment (adidahiya) [inline packages/select/package.json]: use the new `sass-compile` script which gilad recently added (pull in master, see the other packages)

File: packages/select/src/blueprint-select.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+// Licensed under the terms of the LICENSE file distributed with this project.
```
Comment (adidahiya) [inline packages/select/src/blueprint-select.scss]: weird start of line comment syntax

File: packages/select/src/components/_index.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+// Licensed under the terms of the LICENSE file distributed with this project.
```
Comment (adidahiya) [inline packages/select/src/components/_index.scss]: same here, comment syntax

File: packages/select/src/components/omnibar/omnibar.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ import {
     Utils,
 } from "@blueprintjs/core";
 
-import * as Classes from "../../common/classes";
+import * as Classes from "../classes";
```
Comment (adidahiya) [inline packages/select/src/components/omnibar/omnibar.tsx]: I think the previous path was fine...? a `common/` folder next to `components/`?

File: yarn.lock [hunk 6]
```diff
@@ -708,6 +708,10 @@ boom@5.x.x:
   dependencies:
     hoek "4.x.x"
 
+bourbon@^4.3.4:
```
Comment (adidahiya) [inline yarn.lock]: there should be no lockfile diffs. make sure you pull in master

File: packages/select/src/components/omnibar/omnibar.tsx (hunk 5 above)
Comment (hellochar) [reply packages/select/src/components/omnibar/omnibar.tsx]: there's nothing else in common; do we need another directory level?

File: yarn.lock (hunk 6 above)
Comment (hellochar) [reply yarn.lock]: I just did, tried reverting yarn.lock to the master version, ran `yarn`, and these came back?

File: packages/select/package.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  "dependencies": {
+    "@blueprintjs/core": "^1.34.0",
+    "classnames": "^2.2",
+    "pure-render-decorator": "^1.2.1",
+    "tslib": "^1.5.0"
+  },
+  "devDependencies": {
+    "@blueprintjs/karma-build-scripts": "^0.2.0",
+    "@blueprintjs/node-build-scripts": "^0.2.0",
+    "bourbon": "^4.3.4",
```
Comment (adidahiya) [inline packages/select/package.json]: remove this

File: packages/select/package.json [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    "karma": "^1.7.1",
+    "mocha": "^4.0.1",
+    "node-sass": "^4.7.2",
+    "npm-run-all": "^4.1.1",
+    "react": "^15.6.1",
+    "react-addons-css-transition-group": "^15.6.1",
+    "react-dom": "^15.6.1",
+    "react-test-renderer": "^15.6.1",
+    "tslint": "^5.8.0",
+    "typescript": "~2.4.2",
```
Comment (adidahiya) [inline packages/select/package.json]: bump to `~2.6.2`

File: packages/select/src/components/omnibar/omnibar.tsx (hunk 5 above)
Comment (adidahiya) [reply packages/select/src/components/omnibar/omnibar.tsx]: it's just nice to have the convention across packages

**Pull request merged**